unit/memory-analyzer/gdb_api.cpp

```diff
@@ -0,0 +1,208 @@
+/*******************************************************************\
+
+Module: GDB Machine Interface API unit tests
+
+Author: Malte Mues <mail.mues@gmail.com>
+        Daniel Poetzl
+
+\*******************************************************************/
+
+#include <testing-utils/use_catch.h>
+
+// clang-format off
+#if defined(__linux__) || \
+    defined(__FreeBSD_kernel__) || \
+    defined(__GNU__) || \
+    defined(__unix__) || \
+    defined(__CYGWIN__) || \
+    defined(__MACH__)
+#define RUN_GDB_API_TESTS
+#endif
+// clang-format on
+
+#ifdef RUN_GDB_API_TESTS
+
+#include <cstdio>
+#include <cstdlib>
+#include <regex>
+#include <string>
+
+#include <fstream>
+#include <iostream>
+
+#include <memory-analyzer/gdb_api.cpp>
+
+void compile_test_file()
+{
+  std::string test_file("memory-analyzer/test.c");
+
+  std::string cmd("gcc -g -o test ");
+  cmd += test_file;
+
+  const int r = system(cmd.c_str());
+  REQUIRE(!r);
+}
+
+class gdb_api_testt : public gdb_apit
+{
+  explicit gdb_api_testt(const char *binary) : gdb_apit(binary)
+  {
+  }
+
+  friend void gdb_api_internals_test();
+};
+
+void gdb_api_internals_test()
+{
+  compile_test_file();
+
+  SECTION("parse gdb output record")
+  {
+    gdb_api_testt gdb_api("test");
+
+    gdb_api_testt::gdb_output_recordt gor;
+
+    gor = gdb_api.parse_gdb_output_record(
```

Comment:
:pick: declare and initialise in one step (also a bunch of other places)

Reply:
Done.